pkgs/development/libraries/criterion/default.nix

```diff
@@ -1,26 +1,31 @@
-{ lib, stdenv, fetchFromGitHub, boxfort, cmake, libcsptr, pkg-config, gettext
-, dyncall , nanomsg, python3Packages }:
+{ lib, stdenv, fetchFromGitHub, boxfort, meson, libcsptr, pkg-config, gettext
+, cmake, ninja, protobuf, libffi, libgit2, dyncall, nanomsg, nanopb, python3Packages, git }:
 
-stdenv.mkDerivation rec {
-  version = "2.3.3";
+let
+  nanopbMalloc = nanopb.override { mallocBuild = true; };
+in stdenv.mkDerivation rec {
+  version = "2.4.1";
   pname = "criterion";
 
   src = fetchFromGitHub {
     owner = "Snaipe";
     repo = "Criterion";
     rev = "v${version}";
-    sha256 = "0y1ay8is54k3y82vagdy0jsa3nfkczpvnqfcjm5n9iarayaxaq8p";
+    sha256 = "KT1XvhT9t07/ubsqzrVUp4iKcpVc1Z+saGF4pm2RsgQ=";
     fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ meson ninja cmake pkg-config protobuf git ];
 
   buildInputs = [
     boxfort.dev
     dyncall
     gettext
     libcsptr
     nanomsg
+    nanopbMalloc
```

Comment:
```suggestion
    nanopb
```

Reply:
Regular nanopb is missing required symbols.

Reply:
This is in addition to moving it to top-level or we just enable the flag by default in nanopb.

Reply:
I have considered enabling the flag by default, but that runs the risk of a rebuild cascade + potential memory issues in apps that don't need it, so I'll probably just add `nanopbMalloc` to top-level

Reply:
> that would cause unnecessary rebuilds

Thats fine.

> memory issues in apps that don't need it

That wouldn't be that great.

> so I'll probably just add nanopbMalloc to top-level

I don't think we need that. We can just enable that one flag in an override for criterion.
Basically we move the override to top-level and keep the library name so that we do not use the normal library by accident.

Reply:
I think it is fine. tltr: if malloc is not supported then it won't alloc.

<details>
<summary>my dive-deep</summary>

```
$ git checkout 0.4.6.4
$ git grep -rl PB_ENABLE_MALLOC '*.[ch]'
extra/pb_syshdr.h
pb.h
pb_decode.c
pb_decode.h
tests/alltypes_callback/decode_alltypes_callback.c
tests/alltypes_proto3_callback/decode_alltypes_callback.c
tests/decode_unittests/decode_unittests.c
```

I have ignored test files since not distributed.

extra/pb_syshdr.h:
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/extra/pb_syshdr.h#L64-L72
only prototype

pb.h:
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/pb.h#L14 comment
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/pb.h#L90-L92 prototype
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/pb.h#L473-L480 prototype

pb_decode.c:
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/pb_decode.c#L45-L50 prototype
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/pb_decode.c#L621-L625 fail since no malloc support
other place
The other place is calling release if malloc is supported.

pb_decode.h:
https://github.com/nanopb/nanopb/blob/afc499f9a410fc9bbf6c9c48cdd8d8b199d49eb4/pb_decode.h#L110-L119 prototype
</details>

Reply:
Enabling or disabling `PB_ENABLE_MALLOC` makes no functional difference if the `.proto` or `.options` files do not specify fields to have type `FT_POINTER`. It only has small code size and performance differences.